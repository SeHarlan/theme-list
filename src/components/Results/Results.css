.Results {
  height: 100vh;
  nav {
    text-align: center;
    padding: 1rem;
    display: flex;
    justify-content: center;
    align-items: center;
  }
  button {
    box-sizing: content-box;
    height: 1.5rem;
    padding: 0 1rem;
    margin: 0 1rem;
    border-radius: 1rem;
    outline: none;
    font-family: inherit;
    color: inherit;
    background-color: inherit;
    font-size: 1rem;
  }
  div {
    height: 2rem;
  }
  ul {
    margin: 1rem 10vw;
    height: 60vh;
    overflow: scroll;
    text-align: center;
  }
  li {
    list-style: none;
    margin: 0 10vw 3rem 10vw;
  }
  button:hover {
    box-shadow: 2px 2px 2px grey;
    transform: translate(0, -2px);
    cursor: pointer;
  }
  button:active {
    box-shadow: 0 0 0;
    transform: translate(0, 0px);
  }
  button:disabled {
    box-shadow: 0 0 0;
    transform: translate(0, 0);
    cursor: inherit;
  }
}