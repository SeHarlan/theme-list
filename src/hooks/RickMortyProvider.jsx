import React, { createContext, useState, useEffect, useContext } from 'react';
import PropTypes from 'prop-types';
import { fetchCharacters } from '../services/rickMorty';

const RickMortyContext = createContext();

export const RickMortyProvider = ({ children }) => {
  const [aliveCharacters, setAliveCharacters] = useState([]);
  const [deadCharacters, setDeadCharacters] = useState([]);
  const [page, setPage] = useState(1);
  const [totalPages, setTotalPages] = useState(25);
  const [theme, setTheme] = useState('Alive');

  const incPage = () => setPage(prev => (prev + 1));
  const decPage = () => setPage(prev => (prev - 1));

  useEffect(() => {
    fetchCharacters(page)
      .then(({ alive, dead, totalPages }) => {
        setAliveCharacters(alive);
        setDeadCharacters(dead);
        setTotalPages(totalPages);
      });
  }, []);

  return (
    <RickMortyContext.Provider value={{
      aliveCharacters, 
      deadCharacters, 
      theming: { theme, setTheme }, 
      paging: { decPage, incPage, page, totalPages }
    }}
    >
      {children}
    </RickMortyContext.Provider>
  );
};

RickMortyProvider.propTypes = { children: PropTypes.node };

export const useCharacters = () => {
  const { aliveCharacters, deadCharacters } = useContext(RickMortyContext);
  return { aliveCharacters, deadCharacters }; 
};

export const usePaging = () => {
  const { paging } = useContext(RickMortyContext);
  return paging;
};
export const useTheming = () => {
  const { theming } = useContext(RickMortyContext);
  return theming;
};
