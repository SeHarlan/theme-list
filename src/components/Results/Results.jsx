import React from 'react';
import { useTheming, useCharacters, usePaging } from '../../hooks/RickMortyProvider';
import withList from '../../HOCs/WithList';
import Character from '../Character/Character';
import themeStyle from '../../Theme.css';
import style from './Results.css';

const Results = () => {
  const { theme } = useTheming();
  const { deadCharacters, aliveCharacters } = useCharacters();
  const { page, totalPages, decPage, incPage } = usePaging();
  const themeToUse = (theme === 'Alive') ? themeStyle.AliveTheme : themeStyle.DeadTheme;

  const list = (theme === 'Alive') ? aliveCharacters : deadCharacters;
  const Characters = withList(Character);

  return (
    <section className={style.Results + ' ' + themeToUse}>
      <nav>
        <button disabled={page === 1} onClick={decPage}>Previous Page</button>
        <p>{page}/{totalPages}</p>
        <button disabled={page === totalPages} onClick={incPage}>Next Page</button>
      </nav>
      <Characters list={list} />
    </section>
  );
};


export default Results;
