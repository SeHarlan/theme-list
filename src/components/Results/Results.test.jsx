import React from 'react';
import { shallow } from 'enzyme';
import Results from './Results';
import { RickMortyProvider } from '../../hooks/RickMortyProvider';

describe('Results component', () => {
  it('renders Results', () => {
    const wrapper = shallow(
      <RickMortyProvider>
        <Results/>
      </RickMortyProvider>);
    expect(wrapper).toMatchSnapshot();
  });
});
