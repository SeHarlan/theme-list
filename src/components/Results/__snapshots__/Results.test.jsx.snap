// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results component renders Results 1`] = `
<ContextProvider
  value={
    Object {
      "aliveCharacters": Array [],
      "deadCharacters": Array [],
      "paging": Object {
        "decPage": [Function],
        "incPage": [Function],
        "page": 1,
        "totalPages": 25,
      },
      "theming": Object {
        "setTheme": [Function],
        "theme": "Alive",
      },
    }
  }
>
  <Results />
</ContextProvider>
`;
